dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/JdbcTeiAnalyticsTableManager.java

```diff
@@ -288,6 +289,36 @@ private List<AnalyticsTableColumn> getColumns(
     return columns;
   }
 
+  /**
+   * Returns the select clause, potentially with a cast statement, based on the given value type.
+   *
+   * @param valueType the value type to represent as database column type.
+   */
+  private String getSelectClause(ValueType valueType, String columnName) {
```

Comment:
We can change the method name, to `castBasedOnType`, or something similar...
It does not really return a "select".